app/models/chargeback/report_options.rb

```diff
@@ -0,0 +1,102 @@
+class Chargeback
```

Comment:
You can re-open the class any time. Having the `class Chargeback` here is the same as putting the `ReportOptions` block into the `app/models/chargeback.rb`.

The `ReportOptions` is basically new Datatype. That is what `Struct.new` does, the new datatype.

Having a Struct here has great advantage. Struct is basically hash on steroids. A hash can be replaces with a Struct (struct defines the same methods) and the upside is that we can define methods on the given struct.

I have used this patter multiple times in the `ui`. It helped to lighten up some of the controllers. These controllers were previously manipulated complex hashes. Now, they just issue a method on the struct to do the right thing.